Signal/src/view controllers/CountryCodeViewController.m

```diff
@@ -18,10 +18,69 @@ @implementation CountryCodeViewController
 - (void)viewDidLoad {
     [super viewDidLoad];
     [self.navigationController.navigationBar setTranslucent:NO];
-    _countryCodes = [PhoneNumberUtil countryCodesForSearchTerm:nil];
+    _countryCodes = [self countryCodesForSearchTerm:nil];
     self.title    = NSLocalizedString(@"COUNTRYCODE_SELECT_TITLE", @"");
 }
 
+- (NSArray *)countryCodesForSearchTerm:(NSString *)searchTerm {
+    NSMutableArray *result = [NSMutableArray array];
+    for (NSString *countryCode in [PhoneNumberUtil countryCodesForSearchTerm:nil]) {
+        NSString *callingCode = [self callingCodeFromCountryCode:countryCode];
+        if (callingCode != nil &&
+            ![callingCode isEqualToString:@"+0"]) {
+            [result addObject:countryCode];
+        }
+    }
+    return result;
+}
+
+- (NSString *)callingCodeFromCountryCode:(NSString *)countryCode {
+    NSString *callingCode = [PhoneNumberUtil callingCodeFromCountryCode:countryCode];
+    if ([countryCode isEqualToString:@"AQ"]) {
```

Comment:
`PhoneNumberUtil` uses a third-party library `NBPhoneNumberUtil`, which is returning bad data for some obscure country codes.